structured/structured.proto

```diff
@@ -59,14 +59,15 @@ message IndexDescriptor {
   optional string name = 1 [(gogoproto.nullable) = false];
   optional uint32 id = 2 [(gogoproto.nullable) = false, (gogoproto.customname) = "ID"];
   optional bool unique = 3 [(gogoproto.nullable) = false];
+  optional bool primary = 4 [(gogoproto.nullable) = false];
```

Comment:
Rather than a `primary` bool here (which allows for the possibility of multiple indexes being marked `primary`), I think it would be better for `TableDescriptor` to have an `optional IndexDescriptor primary` field which is `nullable = false`. Then you are assured there is one primary key for the table and that it is easily accessible. The various loops over the secondary keys can use `indexes` instead of `indexes[1:]` as they are currently doing.